### ISSUE TITLE
Media optimizer

### ISSUE THREAD

**Issue by Maxou44** [UnicornTranscoder/UnicornTranscoder#4]
The media optimizer allow the server administrator to optimize files to improve performances on a specific device (TV / Mobile / Web) for example.

Actually Plex call the fake FFMPEG but the transcode is broken.

**Comment by DrEVILish** [comment]
Because the media optimiser isn't a priority in contrast to streaming.

Add a detector in UniconFFMPEG to detect if the requested transcode is PLEX Stream or PLEX Optimiser, this would allow you to fall back to the original PLEX transcoder.

Instead on install of deleting the PLEX_transcoder (can't remember the actual name) mv it to PLEX_local_transcoder and trigger that from UniconFFMPEG if it's a PLEX Optimiser run.

PLEX Optimiser only allows one instance to run afaik with the transcoder. Can set this to slower this would also fix the Sync issue.

> Background transcoding x264 preset  
> The x264 preset value used for background transcoding (Sync and Media Optimizer). Slower values will result in better video quality and smaller file sizes, but will take significantly longer to complete processing.

**Comment by Maxou44** [comment]
Implementation idea:

- The load balancer catches the request
- The load balancer choose a transcoder
- The load balancer call the transcoder on a specified route with FFMPEG parameters (POST /api/transcode ?)
- The transcoder processs the file and output it in a temporary directory

When it's finish we have 2 cases :
Case 1 (Transcoder have access to the file system) 
- The transcoder moves the file in the right directory

Case 2 (Transcoder don't have acess to the file system)
- The transcoder open a route where the file can be downloaded (GET /api/transcode/:sessionId ?)
- The transcoder send a call to the loadbalancer
- The load balancer download the file and put it in the folder

When the moving ends, it is require to send a call to Plex (not sure if it's a progress 100% from FFMPEG or another call)

**Comment by Maxou44** [comment]
Plex makes a PUT HTTP call on 
`/playlists/1/items?Item%5Btype%5D=42&Item%5Btitle%5D=Isn%27t%20It%20Romantic&Item%5Btarget%5D=&Item%5BtargetTagID%5D=1&Item%5BlocationID%5D=-1&Item%5BLocation%5D%5Buri%5D=library%3A%2F%2Fb1a7b352-7912-4dce-a5c6-cdde49a2251f%2Fitem%2F%252Flibrary%252Fmetadata%252F330003&Item%5BPolicy%5D%5Bscope%5D=all&Item%5BPolicy%5D%5Bvalue%5D=&Item%5BPolicy%5D%5Bunwatched%5D=0&X-Plex-Product=Plex%20Web&X-Plex-Version=3.87.3&X-Plex-Client-Identifier=36p1kth14e6bhq37whdobljp&X-Plex-Platform=Chrome&X-Plex-Platform-Version=72.0&X-Plex-Sync-Version=2&X-Plex-Device=Windows&X-Plex-Device-Name=Chrome&X-Plex-Device-Screen-Resolution=1920x937%2C1920x1080&X-Plex-Token=XXXXXXXXXXXX&X-Plex-Language=en"`

Playlist 1 is the "Background Processing List"

The endpoint returns something like that : 
```
<?xml version="1.0" encoding="UTF-8"?>
<MediaContainer size="1">
<Playlist ratingKey="1" key="/playlists/1/items" guid="com.plexapp.agents.none://55d3184c-28b9-43ce-842c-b7b03ca00e0f" type="playlist" title="Background Processing List" summary="" playlistType="backgroundProcessing" addedAt="1467987859">
</Playlist>
</MediaContainer>
```

If I'm right, "55d3184c-28b9-43ce-842c-b7b03ca00e0f" is the session id of the stream.

The loadbalancer need to catches the PUT /playlists/ endpoint, forward it to plex and see if it can extract the session id, Plex will start the FFMPEG and LoadBalancer will get arguments (and because we know it's a media optimization we can trigger the transcoder)

**Comment by Maxou44** [comment]
Available in 2.2.0 pre-release
